python_modules/dagster/dagster/_core/definitions/asset_daemon_context.py

```diff
@@ -78,7 +78,11 @@ def __init__(
 
         # fetch some data in advance to batch some queries
         self.instance_queryer.prefetch_asset_records(
-            [key for key in self.target_asset_keys if not self.asset_graph.is_source(key)]
+            [
+                key
+                for key in self.target_asset_keys_and_parents
```

Comment:
we were underfetching the asset records we'd actually need, as we always analyze the parents of our target keys.